operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -982,3 +983,7 @@ func allListTypes(c client.Client) ([]client.ObjectList, error) {
 	}
 	return types, nil
 }
+
+func (r *RedpandaReconciler) IsFluxEnabled(useFlux *bool) bool {
+	return ptr.Deref(useFlux, !r.UseFluxDefault)
```

Comment:
I think the naming of the field is a bit off and has caused me to see this code as wrong at least twice when I don't think it is. Walking through this.

So from the flag, we have:

```
--force-defluxed-mode (passing should set the default of "useFlux" to "false")
```

That gets passed directly to `UseFluxDefault`, so:

```
... --use-flux-default
```

should result in `r.UseFluxDefault == true`.

That would mean that this condition is correct, since it means in the absence of the `UseFlux` pointer field being set, that the function `IsFluxEnabled` returns `false` as it gives `!r.UseFluxDefault`.

I think this would be a lot more clear if we just renamed `r.UseFluxDefault` to `r.DefaultDisableFlux` or some variant of that. Right now `UseFlux*` seems to indicate to me "use flux by default" when this logic is actually the opposite of that.

Reply:
I agree. I changed it.